### PR TITLE
add ubest/vn 68 keyboard

### PR DESCRIPTION
**Pull request by mfkiiyd**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ubest/vn/config.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#ifdef RGB_DI_PIN
+    #define RGBLED_NUM 8
+    #define RGBLIGHT_HUE_STEP 8
+    #define RGBLIGHT_SAT_STEP 8
+    #define RGBLIGHT_VAL_STEP 8
+    #define RGBLIGHT_LIMIT_VAL 240 /* The maximum brightness level */
+    #define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+/*== all animations enable ==*/
+//    #define RGBLIGHT_ANIMATIONS
+// /*== or choose animations ==*/
```
Comment (waffle87) [inline keyboards/ubest/vn/config.h:54]:
```suggestion
```

File: keyboards/ubest/vn/vn.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    K20,    K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C,        K2E,   \
+    K30,  K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C,     K3D, K3E,   \
+    K40,  K41,  K43,  K47,                       K48, K49, K4A, K4B, K4C, K4D, K4E    \
+) { \
+    { K00,   K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,   K0E   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D,   K1E   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,   K2E   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   K3E   }, \
+    { K40,   K41,   KC_NO, K43,   KC_NO, KC_NO, KC_NO, K47,   K48,   K49,   K4A,   K4B,   K4C,   K4D,   K4E   }, \
+}
```
Comment (waffle87) [inline keyboards/ubest/vn/vn.h:50]:
```suggestion
```

File: keyboards/ubest/vn/rules.mk [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+
+LAYOUTS = tkl_ansi
```
Comment (waffle87) [inline keyboards/ubest/vn/rules.mk:20]:
```suggestion
```

File: keyboards/ubest/vn/readme.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+An in-switch RGB LED keyboard with RGB underglow.
+
+* Keyboard Maintainer: [water](https://github.com/mfkiiyd)
+* Hardware Supported: vn
+* Hardware Availability: ZA08A0U0-68KEY V1.0
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ubest/vn:default
+
```
Comment (waffle87) [inline keyboards/ubest/vn/readme.md:12]:
```suggestion

Flashing example for this keyboard:

    make ubest/vn:default:flash
```

File: keyboards/ubest/vn/readme.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (waffle87) [inline keyboards/ubest/vn/readme.md:13]: Please add instructions on entering bootloader mode.
eg.

```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

File: keyboards/ubest/vn/info.json [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+				{"x":3.75, "y":4, "w":5.25},
+				{"x":9, "y":4},
+				{"x":10, "y":4},
+				{"x":11, "y":4},
+				{"x":12, "y":4},
+				{"x":13, "y":4},
+				{"x":14, "y":4},
+				{"x":15, "y":4}
+            ]
+        }
```
Comment (waffle87) [inline keyboards/ubest/vn/info.json:172]:
```suggestion
        }
```

File: keyboards/ubest/vn/info.json [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    "usb": {
+        "vid": "0x6D66",
+        "pid": "0x0868",
+        "device_version": "0.0.1"
+    },
+    "layout_aliases": {
+        "LAYOUT": "LAYOUT_all"
+    },
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (waffle87) [inline keyboards/ubest/vn/info.json:15]: `LAYOUT_all` is generally used with keyboards support multiple layouts.
```suggestion
    "layouts": {
        "LAYOUT": {
```

File: keyboards/ubest/vn/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+	[1] = LAYOUT_all(
+        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  KC_DEL,  _______,
+        _______,     _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
+        _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,
+        _______,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,        _______, _______,
+        _______,   _______,   _______,   _______,                                        _______, _______, _______, _______, _______, _______, _______
+    ),
+	[2] = LAYOUT_all(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,     _______, _______, _______, RESET,   _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
```
Comment (waffle87) [inline keyboards/ubest/vn/keymaps/default/keymap.c:36]:
```suggestion
	[0] = LAYOUT(
        KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_HOME,
        KC_TAB,      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,      KC_PGUP,
        KC_CAPS,       KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,              KC_PGDN,
        KC_LSFT,   KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,        KC_UP,   MO(1),
        KC_LCTL,   KC_LGUI,   KC_LALT,   KC_SPC,                                         KC_RALT, KC_RGUI, KC_APP,  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
    ),
	[1] = LAYOUT(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  KC_DEL,  _______,
        _______,     _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
        _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,
        _______,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,        _______, _______,
        _______,   _______,   _______,   _______,                                        _______, _______, _______, _______, _______, _______, _______
    ),
	[2] = LAYOUT(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______,     _______, _______, _______, QK_BOOT,   _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
```

File: keyboards/ubest/vn/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        _______,   _______,   _______,   _______,                                        _______, _______, _______, _______, _______, _______, _______
+    ),
+	[2] = LAYOUT_all(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______,     _______, _______, _______, RESET,   _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
+        _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,
+        _______,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,        _______, _______,
+        _______,   _______,   _______,   _______,                                        _______, _______, _______, _______, _______, _______, _______
+    ),
+	[3] = LAYOUT_all(
```
Comment (waffle87) [inline keyboards/ubest/vn/keymaps/via/keymap.c:41]:
```suggestion
	[0] = LAYOUT(
        KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC, KC_HOME,
        KC_TAB,      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,      KC_PGUP,
        KC_CAPS,       KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,              KC_PGDN,
        KC_LSFT,   KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,        KC_UP,   MO(1),
        KC_LCTL,   KC_LGUI,   KC_LALT,   KC_SPC,                                         KC_RALT, KC_RGUI, KC_APP,  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
    ),
	[1] = LAYOUT(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  KC_DEL,  _______,
        _______,     _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
        _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,
        _______,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,        _______, _______,
        _______,   _______,   _______,   _______,                                        _______, _______, _______, _______, _______, _______, _______
    ),
	[2] = LAYOUT(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______,     _______, _______, _______, QK_BOOT,   _______, _______, _______, _______, _______, _______, _______, _______, _______,      _______,
        _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,             _______,
        _______,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,        _______, _______,
        _______,   _______,   _______,   _______,                                        _______, _______, _______, _______, _______, _______, _______
    ),
	[3] = LAYOUT(
```

File: keyboards/ubest/vn/vn.h [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define XXX KC_NO
+
+#define LAYOUT_all( \
```
Comment (waffle87) [inline keyboards/ubest/vn/vn.h:23]:
```suggestion
#define LAYOUT( \
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by mfkiiyd** [comment]
Please merge, thank you

**Review by drashna — APPROVED** [review]

File: keyboards/ubest/vn/readme.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+* Hardware Supported: vn
+* Hardware Availability: ZA08A0U0-68KEY V1.0
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ubest/vn:default
+
+Flashing example for this keyboard:
+
+    make ubest/vn:default:flash
```
Comment (fauxpark) [inline keyboards/ubest/vn/readme.md:15]:
```suggestion
    make ubest/vn:default:flash

```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**